modules/__tests__/requestToApi.test.js

```diff
@@ -63,4 +63,97 @@ describe('requestToApi', () => {
       expect(response.ok).toBeTruthy()
     })
   })
+
+  it('should run onIntercept when api call is failing with status 401', () => {
+    const mockFailingXHR = {
+      getAllResponseHeaders: jest.fn(),
+      onload: jest.fn(),
+      open: jest.fn(),
+      onreadystatechange: jest.fn(),
+      readyState: 4,
+      responseText: JSON.stringify({message: 'Credentials are required to access this path'}),
+      send: jest.fn(),
+      status: 401,
+      setRequestHeader: jest.fn(),
+    }
+    const mockResetXHR = {
+      getAllResponseHeaders: jest.fn(),
+      onload: jest.fn(),
+      open: jest.fn(),
+      onreadystatechange: jest.fn(),
+      readyState: 4,
+      responseText: JSON.stringify({newToken: 'abc123456'}),
+      send: jest.fn(),
+      status: 201,
+      setRequestHeader: jest.fn(),
+    }
+    window.XMLHttpRequest = jest.fn(() =>mockFailingXHR);
+    const onIntercept = jest.fn(({currentParams, request, status}) => {
+      if (status === 401) {
+        window.XMLHttpRequest = jest.fn(() => mockResetXHR);
+        const resetRequest = requestToApi({
+          url: 'https://api.github.com/auth/reset',
+          method: 'POST',
+          body: {
+            resetToken: 'abcxzz'
+          }
+        })
+        mockResetXHR.onreadystatechange();
+        resetRequest.then((result) => {
+          const response = result.data.newToken;
+          expect(response).toBe('abc123456');
+        });
+        window.XMLHttpRequest = jest.fn(() => mockXHR);
```

Comment:
I don't want to reset to the `oldXMLHttpRequest` since I still want to use the original `mockXHR` so that after the interceptor is fired, `requestToApi` will be able to get the mocked message

Reply:
OK got 👍